spring-cloud-aws-messaging/src/main/java/io/awspring/cloud/messaging/listener/AbstractMessageListenerContainer.java

```diff
@@ -266,13 +267,22 @@ public boolean isActive() {
 	public void afterPropertiesSet() throws Exception {
 		validateConfiguration();
 		initialize();
+		validateFifoConfiguration();
 	}
 
 	private void validateConfiguration() {
 		Assert.state(this.amazonSqs != null, "amazonSqs must not be null");
 		Assert.state(this.messageHandler != null, "messageHandler must not be null");
 	}
 
+	private void validateFifoConfiguration() {
+		if (getRegisteredQueues().values().stream().anyMatch(queueAttributes -> queueAttributes.isFifo())
+				&& getAmazonSqs() instanceof AmazonSQSBufferedAsyncClient) {
+			getLogger().warn(
+					"`AmazonSQSBufferedAsyncClient` that Spring Cloud AWS uses by default to communicate with SQS is not compatible with FIFO queues.");
```

Comment:
Perhaps we could already suggest a solution here
```suggestion
					"`AmazonSQSBufferedAsyncClient` that Spring Cloud AWS uses by default to communicate with SQS is not compatible with FIFO queues.");
```
```suggestion
					"AmazonSQSBufferedAsyncClient that Spring Cloud AWS uses by default to communicate with SQS is not compatible with FIFO queues. Consider registering non-buffered AmazonSQSAsyncClient bean.");
```